luaui/Widgets/Shaders/DrawPrimitiveAtUnit.geom.glsl

```diff
@@ -82,55 +82,60 @@ void main(){
-			offsetVertex4( width * 0.5, 0.0, -length * 0.5, 0.0, 0.0);
-			offsetVertex4(-width * 0.5, 0.0,  length * 0.5, 1.0, 1.0);
-			offsetVertex4(-width * 0.5, 0.0, -length * 0.5, 1.0, 0.0);
+		if (numVertices == 4u){ // A quad
+			offsetVertex4( width * 0.5, 0.0,  length * 0.5, 0.0, 1.0, 1.414);
+			offsetVertex4( width * 0.5, 0.0, -length * 0.5, 0.0, 0.0, 1.414);
+			offsetVertex4(-width * 0.5, 0.0,  length * 0.5, 1.0, 1.0, 1.414);
+			offsetVertex4(-width * 0.5, 0.0, -length * 0.5, 1.0, 0.0, 1.414);
 			EndPrimitive();
 		}
 	#endif
-	
+
 	#ifdef USE_CORNERRECT
-		if (numVertices == uint(2)){ // A quad with chopped off corners
+		if (numVertices == 2u){ // A quad with chopped off corners
 			float csuv = (cs / (length + width))*2.0;
-			offsetVertex4( - width * 0.5 , 0.0,  - length * 0.5 + cs, 0, csuv); // bottom left
-			offsetVertex4( - width * 0.5 , 0.0,  + length * 0.5 - cs, 0, 1.0 - csuv); // top left
-			offsetVertex4( - width * 0.5 + cs, 0.0,  - length * 0.5 , csuv, 0); // bottom left
-			offsetVertex4( - width * 0.5 + cs, 0.0,  + length * 0.5, csuv, 1.0); // top left
-			offsetVertex4( + width * 0.5 - cs, 0.0,  - length * 0.5 , 1.0 - csuv, 0.0); // bottom right
-			offsetVertex4( + width * 0.5 - cs, 0.0,  + length * 0.5 ,1.0 - csuv, 1.0 ); // top right
-			offsetVertex4( + width * 0.5 , 0.0,  - length * 0.5 + cs , 1.0 , csuv ); // bottom right
-			offsetVertex4( + width * 0.5 , 0.0,  + length * 0.5 - cs , 1.0 -csuv , 1.0 ); // top right
+			// FIXME: Need to incorporate corner size to get the best results.
+			// This is a fudge that looks about right for selected buildings, which have corners of 15% for a square.
+			float addRadiusCorrFudge = 1.1;
+			offsetVertex4( - width * 0.5 , 0.0,  - length * 0.5 + cs, 0, csuv, addRadiusCorrFudge); // bottom left
+			offsetVertex4( - width * 0.5 , 0.0,  + length * 0.5 - cs, 0, 1.0 - csuv, addRadiusCorrFudge); // top left
+			offsetVertex4( - width * 0.5 + cs, 0.0,  - length * 0.5 , csuv, 0, addRadiusCorrFudge); // bottom left
+			offsetVertex4( - width * 0.5 + cs, 0.0,  + length * 0.5, csuv, 1.0, addRadiusCorrFudge); // top left
+			offsetVertex4( + width * 0.5 - cs, 0.0,  - length * 0.5 , 1.0 - csuv, 0.0, addRadiusCorrFudge); // bottom right
+			offsetVertex4( + width * 0.5 - cs, 0.0,  + length * 0.5 ,1.0 - csuv, 1.0, addRadiusCorrFudge); // top right
+			offsetVertex4( + width * 0.5 , 0.0,  - length * 0.5 + cs , 1.0 , csuv, addRadiusCorrFudge); // bottom right
+			offsetVertex4( + width * 0.5 , 0.0,  + length * 0.5 - cs , 1.0 -csuv , 1.0, addRadiusCorrFudge); // top right
 			EndPrimitive();
 		}
 	#endif
 	
 	#ifdef USE_CIRCLES
-		if (numVertices > uint(5)) { //A circle with even subdivisions
+		if (numVertices > 5u){ // A circle with even subdivisions
 			numVertices = min(numVertices,62u); // to make sure that we dont emit more than 64 vertices
+			float internalAngle = float(numVertices) * radians(180.0) / float(numVertices);
+			float addRadiusCorr = 1 / sin(internalAngle / 2.0);
```

Comment:
For performance, we could just call this `1.0`. I'm afraid I don't have any idea how these changes will affect performance.